### PR TITLE
ISPN-8159 Deprecate unused 'module' xml config attribute of cache element

### PR DESCRIPTION
**Pull request by anistor**
https://issues.jboss.org/browse/ISPN-8159

### CODE REVIEW

File: server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CacheConfigurationAdd.java [hunk 1]
```diff
@@ -222,6 +222,9 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
         ModelNode resolvedValue;
         final String templateConfiguration = (resolvedValue = CacheConfigurationResource.CONFIGURATION.resolveModelAttribute(context, cacheModel)).isDefined() ? resolvedValue.asString() : null;
         final ModuleIdentifier moduleId = ModelNodes.asModuleIdentifier(CacheConfigurationResource.CACHE_MODULE.resolveModelAttribute(context, cacheModel));
+        if (moduleId != null) {
```
Comment (tristantarrant) [inline server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CacheConfigurationAdd.java]: Better solution: move the warning to https://github.com/infinispan/infinispan/blob/37e74f53bd426354e842dc9a41a9b96cdff8db65/server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/InfinispanSubsystemXMLReader.java#L638 and also i18n it.

**Comment by anistor** [comment]
Done!

**Comment by tristantarrant** [comment]
Temporarily closing because of CI overload

**Comment by anistor** [comment]
Replaced by https://github.com/infinispan/infinispan/pull/5380

**Pull request closed without merging**